buildrunner/validation/config.py

```diff
@@ -132,6 +138,155 @@ def validate_push(
                 if update_mp_push_tags and name is not None:
                     mp_push_tags.add(name)
 
+        def get_source_image(step: Step) -> Optional[List[str]]:
+            """
+            Get the source image from the step.build and/or step.run.image
+
+            Args:
+                step (Step): Build step
+
+            Returns:
+                Optional[str]: Source images
+            """
+            dockerfile_text = None
+            src_image = []
+            is_multi_platform = False
+            dockerfile = None
+
+            # Gets the source image from build.dockerfile
+            if step.build:
+                dockerfile = step.build.dockerfile
+                if not dockerfile and step.build.path:
+                    dockerfile = f"{step.build.path}/Dockerfile"
+
+                if dockerfile:
+                    if os.path.exists(dockerfile):
+                        with open(dockerfile, "r") as file:
+                            dockerfile_text = file.read()
+                    else:
+                        # Dockerfile is defined as a string in the config file
+                        dockerfile_text = dockerfile
+
+                if not dockerfile_text:
+                    raise ValueError(
+                        f"There is an issue with the dockerfile {dockerfile}"
+                    )
+
+                for line in dockerfile_text.splitlines():
+                    if line.startswith("FROM"):
+                        image_name = line.replace("FROM ", "").strip()
```

Comment:
This will probably not catch everything, but it's a start. i.e. `FROM  someimage` with two spaces.

Reply:
Yeah there are some errors here, I am switching this to use `dockerfile_parse` library

Reply:
I reverted back to doing the parsing for the image manually. The `dockerfile_parse` library was overwriting the `Dockerfile` when I was running the tests. There may be a way to use the lib without it doing that but I thought it was probably just as easy to parse it manually.